Documentation/design-docs/DotNetCore-SharedPackageStore.md

```diff
@@ -0,0 +1,216 @@
+# .NET Core - Shared Package Store
+
+## Introduction
+To enable sharing of assemblies among all machine-wide .NET Core applications, a centralized shared assembly store is needed. It enables applications to be trimmed of the shared assemblies and frees the apps from carrying them. This document will describe such an assembly store and its lookup that applications can take advantage of during development and their deployment. It also describes commands that will augment `dotnet` to compose shared package store entries and publishing mechanisms through which apps can filter assemblies to reduce disk usage.
+
+### Packages Store
+
+The package store can be either a global system-wide folder or a user local folder:
+
++ **Global**:
+    - The `dotnet` root location -- on Windows, the folder is located in `C:\Program Files (x86)\`. See layout below.
+    - In the `packages/install` folder we expect packages to be installed ONLY through platform installers like MSI, pkg, deb, apt-get etc.
+    - The package layout composed with the `dotnet cache` command (details follow) are expected to be unzipped directly into the `packages` folder. Note, that this unzip step is a manual action.
+
+```
+    - dotnet.exe
+    - shared
+        - netcoreapp2.0
+            + 2.0.0-preview2-00001
+    - packages
+        - install
+            + refs
+            + netcoreapp2.0
+            + netcoreapp2.1
+        + refs
+        + netcoreapp2.0
+        + netcoreapp2.1  
+```
+
+The layout within `netcoreapp*` folders is a NuGet cache layout.
+
++ **User**: The user local location can be used to override the global location when probing assemblies and uses a very similar layout, except that the `packages/install` folder is NOT supported in the user profile. It is only supported in the global location.
+
+```
+    - %USERPROFILE% or $HOME/
```

Comment:
@gkhanna79 
As per this doc the global location would be like
C:\Users\<user>\.dotnet\x64\packages\netcoreapp1.0\newtonsoft.json\9.0.1\lib\netstandard1.0\Newtonsoft.Json.dll

and additional probe directories will be like : 
[user specified dir]\x64\netcoreapp1.0\newtonsoft.json\9.0.1\lib\netstandard1.0\Newtonsoft.Json.dll

do we want to ensure the packages appear on both and ensure both have the same format  or just the format?
C:\Users\<user>\.dotnet\packages\x64\netcoreapp1.0\newtonsoft.json\9.0.1\lib\netstandard1.0\Newtonsoft.Json.dll
[user specified dir]\x64\netcoreapp1.0\newtonsoft.json\9.0.1\lib\netstandard1.0\Newtonsoft.Json.dll